cpp/src/arrow/compute/kernels/scalar_if_else_benchmark.cc

```diff
@@ -24,42 +24,79 @@
 #include "arrow/testing/gtest_util.h"
 #include "arrow/testing/random.h"
 #include "arrow/util/key_value_metadata.h"
+#include "arrow/util/logging.h"
 
 namespace arrow {
 namespace compute {
 
 const int64_t kNumItems = 1024 * 1024;
 const int64_t kFewItems = 64 * 1024;
 
-template <typename Type, typename Enable = void>
-struct GetBytesProcessed {};
+struct GetBytesProcessedVisitor {
+  const Array* arr;
+  int64_t total_bytes = 0;
 
-template <>
-struct GetBytesProcessed<BooleanType> {
-  static int64_t Get(const std::shared_ptr<Array>& arr) { return arr->length() / 8; }
-};
+  explicit GetBytesProcessedVisitor(const Array* arr) : arr(arr) {}
 
-template <typename Type>
-struct GetBytesProcessed<Type, enable_if_number<Type>> {
-  static int64_t Get(const std::shared_ptr<Array>& arr) {
+  Status RecurseInto(const Array* child_arr) {
+    // return VisitTypeInline(*child_arr->type(), this);
```

Comment:
Did you mean to re-enable this or just to remove this function? `RecurseInto` is currently a no-op...

Reply:
My bad. I commented this to reduce the template error noise when compiling and forgot to re-enable.